client/backoff/backoff.go

```diff
@@ -64,62 +64,51 @@ func (b *Backoff) Reset() {
 // Backoff is not reset by this function.
-func (b *Backoff) Retry(ctx context.Context, f func() error) error {
+func (b *Backoff) Retry(ctx context.Context, f func() error, retry ...codes.Code) error {
 	// If the context is already done, don't make any attempts to call f.
 	if ctx.Err() != nil {
 		return ctx.Err()
 	}
 
 	// Try calling f while the error is retryable and ctx is not done.
 	for {
-		if err := f(); IsRetryable(err) {
-			select {
-			case <-time.After(b.Duration()):
-				continue
-			case <-ctx.Done():
-				return err
-			}
+		if err := f(); !IsRetryable(err, retry...) {
+			return err
+		}
+		select {
+		case <-time.After(b.Duration()):
+		case <-ctx.Done():
+			return ctx.Err()
 		}
-		return nil
 	}
 }
 
-// IsRetryable returns true unless the error is explicitly not retriable per
-// https://godoc.org/google.golang.org/grpc/codes.
-func IsRetryable(err error) bool {
-	switch code := status.Code(err); code {
-
-	// Errors that should not be retried:
-	case codes.OK, // Success = done.
-		codes.InvalidArgument,    // Retrying won't fix an invalid argument.
-		codes.PermissionDenied,   // Retrying won't fix a permission denied error.
-		codes.Unauthenticated,    // Retrying won't fix an unauthenticated error.
-		codes.FailedPrecondition, // Client must wait until the system has been fixed.
-		codes.Unimplemented,      // Retrying won't fix this.
-		codes.Internal,           // Something is very broken.
-		codes.DataLoss,           // Something is very broken.
-		codes.AlreadyExists,      // Retrying won't change this.
-		codes.Canceled:           // The client canceled: stop the retry loop.
-		return false
-
-	// Cases that are debatable:
-	case codes.OutOfRange, // Might be retryable if the server does a lot of work.
-		codes.NotFound: // Retrying might help, but not rapidly.
+// IsRetryable returns false unless the error is explicitly retriable per
+// https://godoc.org/google.golang.org/grpc/codes,
+// or if the error codes is in retry. codes.OK is not retryable.
+func IsRetryable(err error, retry ...codes.Code) bool {
+	code := status.Code(err)
+	switch code {
+	// Fast path.
+	case codes.OK:
 		return false
 
-	// Debatable cases. Retry to preserve existing behavior.
-	case codes.DeadlineExceeded, // The client or the server expired a timeout.
-		codes.ResourceExhausted, // Retry with backoff.
-		codes.Unknown:           // Catch-all error code.
+	// Debatable cases:
+	case codes.DeadlineExceeded,
+		codes.ResourceExhausted: // Retry with backoff.
 		return true
 
 	// Errors that are explicitly retryable:
 	case codes.Unavailable, // Client can just retry the call.
 		codes.Aborted: // Client can retry the read-modify-write function.
 		return true
+	}
 
-	// Retry for all unknown errors to preserve existing behavior.
-	default:
-		return true
+	for _, c := range retry {
+		if code == c {
+			return true
+		}
 	}
+	// Don't retry for all other errors.
+	return false
```

Comment:
This changes the behaviour for normal `error`s so that they are no longer retryable.  As a result, I think that things that use the continuous mode of the ct-go `scanner.Fetcher` (Migrillian and soon also the minimal gossip monitor) are  broken by this.  

https://github.com/google/certificate-transparency-go/pull/393 has my fix/workaround (in the final commit).

Reply:
Yes, I also noticed the change when I was preparing a mirror demo :) I have a fix which I will send shortly (unless you already have one).

Reply:
I fixed it a bit differently by introducing a special ErrRetry in this package, and returning `err == ErrRetry` in this line instead of `false`. Not sure, but perhaps a better option would be to just return `true` here.

Reply:
Maybe have a `RetriableError` type that's just the same as `error`, and allow retry if the error is-a `RetriableError`?  That way, the retryable error could include a message too.

Reply:
I think this is a good idea.